metaflow/runner/subprocess_manager.py

```diff
@@ -9,6 +9,8 @@
 import threading
 from typing import Callable, Dict, Iterator, List, Optional, Tuple
 
+from .utils import check_process_exited
+
 
 def kill_process_and_descendants(pid, termination_timeout):
```

Comment:
I guess we can remove this? If it is not being used, since you replaced it by a plural version..

Actually, it is used in the `kill()` function..
Maybe you can modify `kill` to use your plural version and then get rid of this one perhaps..

If you do it, we can copy over the TODO comments to the plural version..
also, you can use this `kill` in `_handle_sigint` and `_async_handle_sigint`

Reply:
Reimplemented kill in terms of the plural version to keep the API incase someone depends on it.

> also, you can use this kill in _handle_sigint and _async_handle_sigint
As in calling kill in a loop? I'd rather avoid that since we can just spawn one subprocess to send a signal to all managed processes instead of one for each.